### PR TITLE
feat: Public share locked and expired UI

### PR DESCRIPTION
**Pull request by adrien-coye**
- [x] Network stack error parsing
- [x] Public share UI
- [x] Expired UI
- [x] i18n

Feature branch https://github.com/Infomaniak/ios-kDrive/pull/1306

IMG_3394
IMG_3393

### CODE REVIEW

File: kDrive/AppDelegate.swift [hunk 1]
```diff
@@ -107,11 +107,12 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
         Task {
             try! await Task.sleep(nanoseconds:5_000_000_000)
             print("coucou")
-            let somePublicShare = URL(string: "https://kdrive.infomaniak.com/app/share/140946/01953831-16d3-4df6-8b48-33c8001c7981")
-            //await UIApplication.shared.open(somePublicShare!) // opens safari
+            // a public share expired
+            let somePublicShare = URL(string: "https://kdrive.infomaniak.com/app/share/140946/81de098a-3156-4ae6-93df-be7f9ae78ddd")
```
Comment (adrien-coye) [inline kDrive/AppDelegate.swift:111]: We can merge this to the host PR, it does not matter yet. The last PR will do the cleanup.

File: kDrive/AppRouter.swift [hunk 2]
```diff
@@ -585,6 +585,61 @@ public struct AppRouter: AppNavigable {
 
     // MARK: RouterFileNavigable
 
+    @MainActor public func presentPublicShareLocked(_ destinationURL: URL) {
+        guard let window,
```
Comment (adrien-coye) [inline kDrive/AppRouter.swift:589]: The routing will change on the last PR to work offline too.

File: kDrive/AppRouter.swift [hunk 3]
```diff
@@ -585,6 +585,61 @@ public struct AppRouter: AppNavigable {
...
+            guard let navigationController = rootViewController.selectedViewController as? UINavigationController else {
+                return
+            }
+
+            navigationController.present(publicShareNavigationController, animated: true, completion: nil)
+        }
+    }
+
+    @MainActor public func presentPublicShareExpired() {
+        guard let window,
```
Comment (adrien-coye) [inline kDrive/AppRouter.swift:617]: The routing will change on the last PR to work offline too.

File: kDrive/UI/Controller/Files/External/BaseInfoViewController.swift [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import kDriveCore
+import kDriveResources
+import UIKit
+
+class BaseInfoViewController: UIViewController {
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Files/External/BaseInfoViewController.swift:23]: Pure autolayout code view, not added comments to match what is expected.

File: kDrive/UI/Controller/Files/External/LockedFolderViewController.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import kDriveCore
+import kDriveResources
+import UIKit
+
+class LockedFolderViewController: BaseInfoViewController {
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Files/External/LockedFolderViewController.swift:23]: Pure autolayout code view, not added comments to match what is expected.

**Review by adrien-coye — COMMENTED** [review]
I did a self review and left notes for the sake of the next reviewer.

File: kDrive/UI/Controller/Files/External/UnavaillableFolderViewController.swift [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+import kDriveResources
+import UIKit
+
+class UnavaillableFolderViewController: BaseInfoViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        centerImageView.image = KDriveResourcesAsset.ufo.image
+        titleLabel.text = KDriveStrings.Localizable.dropboxPublicShareOutdatedTitle
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Files/External/UnavaillableFolderViewController.swift:27]: public share key

File: kDrive/UI/Controller/Files/External/UnavaillableFolderViewController.swift [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        descriptionLabel.text = KDriveStrings.Localizable.dropboxPublicShareOutdatedDescription
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Files/External/UnavaillableFolderViewController.swift:28]: also public share key

**Review by adrien-coye — COMMENTED** [review]
To change

File: kDriveCore/Utils/Logging.swift
Comment (PhilippeWeidmann) [inline kDriveCore/Utils/Logging.swift:1]: This shouldn't be committed

**Review by PhilippeWeidmann — CHANGES_REQUESTED** [review]
Some general UI remarks:
- [x] Use IKLabel with the correct style instead of .preferredFont & co
- [x] If possible use embed this VC inside a UINavigationViewController to get the native close Button inside the navigationBar
- [x] Spacing is incorrect compared to the figma

(As I wont be there to approve, feel free to remove my request changes after it's done)

File: kDrive/UI/Controller/Files/External/BaseInfoViewController.swift [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        containerView.addSubview(descriptionLabel)
+
+        let views = ["titleLabel": titleLabel,
+                     "descriptionLabel": descriptionLabel,
+                     "centerImageView": centerImageView]
+
+        let verticalConstraints = NSLayoutConstraint
+            .constraints(withVisualFormat: "V:|[centerImageView]-[titleLabel]-[descriptionLabel]|",
+                         metrics: nil,
+                         views: views)
```
[the remark this answers is not in the document]
Comment (adrien-coye) [reply kDrive/UI/Controller/Files/External/BaseInfoViewController.swift:98]: Debatable, and you're not here.

[the remark this answers is not in the document]
Comment (PhilippeWeidmann) [reply kDrive/UI/Controller/Files/External/BaseInfoViewController.swift:87]: The issue with visual constraints is that it's not safe at all. I understand that it "reads" easier but for example if we change the name of centerImage the code will compile fine but it won't work anymore.

From the the top of my head, something like this should work:
```swift
centerImageView.topAnchor.constraint(equalTo: view.topAnchor)
titleLabel.topAnchor.constraint(equalTo: centerImageView.bottomAnchor)
descriptionLabel.topAnchor.constraint(equalTo: titleLabel.bottomAnchor)
descriptionLabel.bottomAnchor.constraint(equalTo: view.bottomAnchor)
```

It is indeed more verbose but it is safe at compile time.

**Review by PhilippeWeidmann — COMMENTED** [review]
Can't test it but I'm fine with merging if everything was addressed.

**Review by PhilippeWeidmann — APPROVED** [review]

**Pull request merged**